src/Functional/functions.php

```diff
@@ -388,12 +388,45 @@ function reThrow(\Exception $e)
  *  liftM2 (+) (Just 1) Nothing = Nothing
  *
  * @param callable $transformation
+ * @param Monad $ma
+ * @param Monad $mb
+ *
+ * @return Monad|\Closure
+ */
+function liftM2(
+    callable $transformation = null,
+    Monad $ma = null,
+    Monad $mb = null
+) {
+    return curryN(
+        3,
+        function (
+            callable $transformation,
+            Monad $ma,
+            Monad $mb
+        ) {
+            return bindM2(function ($a, $b) use ($transformation, $mb): Monad {
```

Comment:
@widmogrod your proposal was very elegant :) but unfortunately `compose` can't handle `$transformation`, since it needs two arguments.